### DISCUSSION TITLE
What does 'create nested one without input' mean and do?

### DISCUSSION THREAD

**Discussion by MincePie** [prisma/prisma#11812] [Q&A] [answered]
I'm trying to find a reference in the documentation for prisma that explains when and why it adds to the client record generated with fields that use the terminology: createNestedOneWithoutInput.

I have seen the [nested writes segment](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-writes), but it seems to have nothing to do with what's happening here.
 
The type generated for a request model and input file I made has been created with additional fields that appear as follows:

  export type RequestCreateInput = {
    
           user: UserCreateNestedOneWithoutRequestInput
  }

I can't find a way to create a record in my db.

The error message when I try says:

_Type 'CreateRequest' is not assignable to type '(Without<RequestCreateInput, RequestUncheckedCreateInput> & RequestUncheckedCreateInput) | (Without<...> & RequestCreateInput)'.
Type 'CreateRequest' is not assignable to type 'Without<RequestUncheckedCreateInput, RequestCreateInput> & RequestCreateInput'. Property 'user' is missing in type 'CreateRequest' but required in type 'RequestCreateInput'.ts(2322) index.d.ts(19184, 5): 'user' is declared here. index.d.ts(8926, 5): The expected type comes from property 'data' which is declared here on type '{ select?: RequestSelect | null | undefined; include?: RequestInclude | null | undefined; data: (Without<...> & RequestUncheckedCreateInput) | (Without<...> & RequestCreateInput); }'_

I don't know what this means, but I think it wants me to define a user. When I try adding one to the input file and then giving the current user id to the prisma create resolver, I get an error (in various formats), in general saying that Type 'typeof User' is not assignable to type 'never'.

I think step 1 is to figure out why this field is being generated and added to my type definition. Can anyone help with a reference to where this terminology is defined and explained? 

Thank you

**Comment by lmwenda** [comment]
im wondering this as well

**Comment by austincrim** [comment]
Hey @MincePie 👋,

It would be easiest if you could share your full Prisma schema and the application code where you are running your queries.

**Reply by cheapestorderconcept** [reply]
I am having this issue too and below is my prisma schema

```
model DropperRequest {
  id             String           @id @unique @default(uuid()) @db.Uuid
  request_title  String
  request_desc   String
  budget         Int
  location       String
  dropperId      String           @db.Uuid
  image          String?
  createdAt      DateTime         @default(now())
  updatedAt      DateTime        @updatedAt
  postedBy       User             @relation(fields: [dropperId], references: [id])
  pickerResponse PickerResponse[]
}
```

**Reply by austincrim** [reply]
Can you share the code that is causing the error?

**Comment by nikkhn** [comment, marked as answer]
@cheapestorderconcept @MincePie 

I was getting this error until I realized I forgot the nested `create` key under the relation I was adding - just in case that was what was your issue too! 

```
  const grant: Array<GrantsType> = [{
    name: "Grant A",
    program: {
        name: "My Program That Could or Could Not Exist Yet"
    }
  }]
```
  
  VS
  
```
const grant: Array<GrantsType> = [{
    name: "Grant A",
    program: {
      create: {
        name: "My Program That Could or Could Not Exist Yet"
      }
    }
  }]
```

**Reply by cheapestorderconcept** [reply]
Yes, I was missing a required field that was why I was getting the error

**Reply by janpio** [reply]
One of you both please create a bug report for this - that error message should be improved so it is easier to understand what is going on.

**Reply by nikkhn** [reply]
@janpio created https://github.com/prisma/prisma/issues/15491

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
